### DISCUSSION TITLE
New API for reading multiple methods from single contract

### DISCUSSION THREAD

**Discussion by nezouse** [wevm/wagmi#1767] [Ideas]
Hello, right now there is only one api to read from contract which allow multiple contract types.
What do you think about another API which allows reading only from single abi?

```js
const {love, play} = useContractRead({address, abi, methods: ['love', 'play']})
```

It may not be that usefull in this case, but still i way prefer object syntax than array syntax.

```js
const [{love, play}, {love, play}] = useContractRead({abi, addresses: ['0x1', '0x2'], methods: ['love', 'play']})
```
And in this case i think it's way easier as now every contract data is encapsulated in object. 

Don't think that's much of a change tbh, but should make working with wagmi easier for a lot of people.

Please share some feedback about it

**Comment by tmm** [comment]
Check out [`useContractReads`](https://wagmi.sh/react/hooks/useContractReads):

```ts
import { useContractReads } from 'wagmi'

const { data, isError, isLoading } = useContractReads({
  contracts: [
    {
      ...wagmigotchiContract,
      functionName: 'love',
      args: ['0x…'],
    },
    {
      ...wagmigotchiContract,
      functionName: 'play',
      args: ['0x…'],
    },
  ],
})
```

**Reply by nezouse** [reply]
Yeah, i know it exists, but it gets tricky when u want to do something like this
import { useContractReads } from 'wagmi'
```js
const { data, isError, isLoading } = useContractReads({
  contracts: [
    {
      ...wagmigotchiContract,
      functionName: 'love',
      args: ['0x…'],
    },
    {
      ...wagmigotchiContract,
      functionName: 'play',
      args: ['0x…'],
    },
        {
      ...wagmigotchiContract1,
      functionName: 'love',
      args: ['0x…'],
    },
    {
      ...wagmigotchiContract1,
      functionName: 'play',
      args: ['0x…'],
    },
  ],
})
```
Here return type would be `['love', 'play', 'love', 'play']`, but in the case im proposing it would be grouped per address so `[{'love', 'play'}, {'love', 'play'}]` which i think is more ergonomic. Sure that's not a big change, but i think it can be a nice addition.
